pkg/doc/verifiable/credential.go

```diff
@@ -619,8 +667,28 @@ func getSchemaLoader(schemas []TypedID, opts *credentialOpts) (gojsonschema.JSON
 	return defaultSchemaLoader, nil
 }
 
-// todo cache credential schema (https://github.com/hyperledger/aries-framework-go/issues/185)
-func loadCredentialSchema(url string, client *http.Client) ([]byte, error) {
+func getJSONSchema(url string, useCache bool, client *http.Client) ([]byte, error) {
+	if !useCache {
+		return loadJSONSchema(url, client)
+	}
+
+	// Check the cache first.
+	if cachedBytes, ok := jsonSchemaCache.get(url); ok {
+		return cachedBytes, nil
+	}
+
+	schemaBytes, err := loadJSONSchema(url, client)
+	if err != nil {
+		return nil, err
+	}
+
+	// Put the loaded schema into cache
+	jsonSchemaCache.put(url, schemaBytes)
```

Comment:
Are you supposed to put it into cache even though useCache is false?

Reply:
One more question: Would we ever _not_ want to use the cache?

Reply:
> Are you supposed to put it into cache even though useCache is false?

It's checked at the beginning of the function:
```
	if !useCache {
		return loadJSONSchema(url, client)
	}
```

Reply:
> One more question: Would we ever _not_ want to use the cache?

I don't know actually :)
By default it's disabled, so maybe it's worth to revert this.
@troyronda what do you think?

Reply:
@sandrask ah, looks like the function returns, so line 686 won't happen.

Reply:
You are right :)